src/core/agent/records/createOrUpdateBasicRecord.ts

```diff
@@ -0,0 +1,11 @@
+import { BasicRecord } from "./basicRecord";
+import { Agent } from "../agent";
+
+export async function createOrUpdateBasicRecord(record: BasicRecord) {
+  const existingRecord = await Agent.agent.basicStorage.findById(record.id);
+  if (existingRecord) {
+    await Agent.agent.basicStorage.update(record);
+  } else {
+    await Agent.agent.basicStorage.save(record);
```

Comment:
After looking at the rest of the PR (and if its not used in signifyNotificationService.ts as I suggest) - it seems like this will be most used for things like setting the favourites.

In this case, it will be set once and then updated forever.

I suggest this to make this faster without adding complexity:
```
try {
  await Agent.agent.basicStorage.update(record);
} catch (error) {
  if (<error is the error for not found in the database>) {
    await Agent.agent.basicStorage.save(record);
  } else {
    throw error;
  }
}
```

and now that I think of, maybe it's easier if this function is just part of the basicStorage class? I don't like parts of the core calling `Agent.agent...` - we should avoid this as it's making our code brittle (I have a story to fix it in other places)